Update CHANGELOG for 10.0.2 (#4772)
citusdata/citus: CHANGELOG.md

### citus v10.0.2 (March 3, 2021) ###

* Fixes insert query with CTEs/sublinks/subqueries etc

* Removes limits around long table names

> * Uses `PROCESS_UTILITY_QUERY` in utility calls

Reviewer: We should do as we did in v9.5.4:
```suggestion
* Fixes a compatibility issue with pg_audit in utility calls
```

Sorry, I should have updated that related pr's description

* Preserves colocation with procedures in `alter_distributed_table`

* Grants `SELECT` permission on `citus_tables` view to `public`

* Fixes a bug with `UNION ALL` on PG 13

> * Grants read access for columnar metadata tables to unprivileged user

Reviewer: Maybe we should rewrite this as:
```suggestion
* Grants `SELECT` permission on columnar metadata tables to `public`
```

As you wrote for `citus_tables`:
>Grants `SELECT` permission on `citus_tables` view to `public`

----

Plus, I think we should group those two entries

* Prevents using `alter_columnar_table_set` and `alter_columnar_table_reset`
  on a columnar table not owned by the user

* Fixes a bug due to cross join without target list

* Improves `citus_update_table_statistics` and provides distributed deadlock
  detection

* Adds a configure flag to enforce security

> 

Reviewer: seems like `Prevent infinite recursion for queries that involve UNION ALL below JOIN` is missing
https://github.com/citusdata/citus/pull/4756

Reviewer: I guess we won't include it in 10.0.3, sorry
